### DISCUSSION TITLE
How to change the output dir for the compiled binary?

### DISCUSSION THREAD

**Discussion by apollo7321** [go-delve/delve#3684] [Q&A] [answered]
I have a git repo with a src folder which contains my Go module (go.mod, main.go, ...). Inside the root folder there is another folder "run" which contains a "workdir" folder which contains config files to run the application in dev mode.

It looks like this:

```
git-project
├───run
│   ├───workdir
│   └───other folders ...
└───src
    ├───go.mod
    ├───main.go
    └───other folders and files ...
```

I want to debug the program/module in src but the binary should be run inside run/workdir, because it uses the binary path to find the configs.

In VS Code I can do that with the following launch.json:
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Launch",
            "type": "go",
            "request": "launch",
            "cwd": "run/workdir",
            "mode": "auto",
            "program": "src",
            "output": "run/workdir/servicebin"
        }
    ]
}
```

It was working a few days ago, but I cannot remember what cli arguments I used. I thought it was like this:

```
cd src
dlv debug --wd ../run/workdir
```

But the paths the program uses are not correct.

**Comment by aarzilli** [comment]
That's the correct option, what does it mean that "the paths the program uses are not correct"?

**Reply by apollo7321** [reply]
I meant the config paths the program uses. For example, it should use run/workdir/service.conf, but it tries to load src/service.conf.

I think I should create an example project, since the program I am trying to debug belongs to my company.

**Reply by apollo7321** [reply]
Okay, I just created an example project and it works as expected. I am not sure what I did wrong at work today. I'll check my dlv setup and such on Monday and will update here if it works. Very strange...

This is my example project: https://github.com/cwansart/godlv-wd

In VS Code:
```
2024/03/08 18:53:15 bin path:     c:\Users\chris\Desktop\godlv\run\workdir\servicebin.exe
2024/03/08 18:53:15 service.conf: workdir
```

On my terminal:
```
chris@apollo-asus MINGW64 ~/Desktop/godlv/src
$ dlv debug --wd ../run/workdir
Type 'help' for list of commands.
(dlv) c
2024/03/08 18:54:11 bin path:     C:\Users\chris\Desktop\godlv\src\__debug_bin735399696.exe
2024/03/08 18:54:11 service.conf: workdir

Process 13156 has exited with status 0
(dlv) exit
```

I am not sure where I messed things up at my work project.

**Reply by apollo7321** [reply]
I just tested it at work on a Linux machine with `dlv debug --wd ../run/workdir` but it does not work. os.Executable() still returns the `src` folder as path. I am really confused.

**Reply by aarzilli** [reply, marked as answer]
Yes, `--wd` doesn't change where the executable is generated (that's the `--output` flag), it changes where it is executed.

**Reply by apollo7321** [reply]
That makes sense. Just tested it, it works. For some reason I did not noticed --output. Thanks.
